### PR TITLE
EZP-29297: 404 error pages for hidden locations should be tagged

### PR DESCRIPTION
**Pull request by kmadejski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-29297](https://jira.ez.no/browse/EZP-29297)
| **Bug**| yes
| **New feature**    | no
| **Target version** | master (`0.7`)
| **BC breaks**      | no
| **Tests pass**     | yes/no
| **Doc needed**     | no

These changes are related to: https://github.com/ezsystems/ezpublish-kernel/pull/2350

### CODE REVIEW

File: src/EventSubscriber/HiddenLocationExceptionSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        if (!$event->getException() instanceof HiddenLocationException) {
+            return;
+        }
+
+        /** @var HiddenLocationException $exception */
+        $exception = $event->getException();
+
+        $response = $this->exceptionController->showAction($event->getRequest(), FlattenException::create($exception));
+        $this->responseTagger->tag($this->responseConfigurator, $response, $exception->getLocation());
```
Comment (andrerom) [inline src/EventSubscriber/HiddenLocationExceptionSubscriber.php]: please rebase thins on top of #67 so you don't need to create response

Comment (andrerom) [reply src/EventSubscriber/HiddenLocationExceptionSubscriber.php]: As in this will also avoid you having to inject _twig.controller.exception_ right?

Comment (kmadejski) [reply src/EventSubscriber/HiddenLocationExceptionSubscriber.php]: Both correct. Thanks to https://github.com/ezsystems/ezplatform-http-cache/pull/67 the solution has been significantly simplified.

**Comment by kmadejski** [comment]
@andrerom  / @adamwojs / @alongosz tests here didn't pass due to dependency on the kernel. This comes from the new spec which I added for this solution. I think we can safely skip forcing composer to load correct kernel branch here and act as if all tests passed 😉.

**Comment by alongosz** [comment]
Control question: does the feature require new kernel or just the tests? 

If it's the former one then [this line](https://github.com/ezsystems/ezplatform-http-cache/blob/master/composer.json#L13) needs to be changed to `"ezsystems/ezpublish-kernel": "^7.2@dev"`, which is probably not what you want. Just asking... :)

**Comment by kmadejski** [comment]
@alongosz this https://github.com/ezsystems/ezplatform-http-cache/pull/69/files#diff-27a7b10b0921b701f0f9be7a6081bcecR18 spec depends on `HiddenLocationException` introduced in: https://github.com/ezsystems/ezpublish-kernel/pull/2350/files#diff-39129cd096d79d4468524079d3dc6fcaR12

If we really want to make this tests pass, then I would need to change the line which you've mentioned to something like `"dev-ezp_29297 as 6.13.2"`. Do you want me to change it to confirm?

**Comment by andrerom** [comment]
@kmadejski can you try to skip this if class is missing?
https://github.com/phpspec/phpspec/pull/296 _(try to do it in let() perhaps)_

**Pull request merged**